flow/record/fieldtypes/__init__.py

```diff
@@ -776,30 +772,33 @@ def __eq__(self, other: object) -> bool:
         if isinstance(other, command):
             return self.executable == other.executable and self.args == other.args
         if isinstance(other, str):
-            return self._join() == other
+            return self._raw == other
         if isinstance(other, (tuple, list)):
             return self.executable == other[0] and self.args == list(other[1:])
 
         return False
 
-    def _split(self, value: str) -> tuple[str, list[str]]:
-        executable, *args = shlex.split(value, posix=self._posix)
-        executable = executable.strip("'\" ")
+    def _split(self, value: str) -> tuple[path, list[str]]:
+        if value:
+            executable, *args = shlex.split(value, posix=self._posix)
+            executable = executable.strip("'\" ")
+        else:
+            executable = ""
+            args = []
 
         return self._path_type(executable), args
 
     def _join(self) -> str:
-        return shlex.join([str(self.executable), *self.args])
+        return self._raw
 
-    def _pack(self) -> tuple[tuple[str, list], str]:
+    def _pack(self) -> tuple[str, int]:
         command_type = TYPE_WINDOWS if isinstance(self, windows_command) else TYPE_POSIX
         if self.executable:
-            _exec, _ = self.executable._pack()
-            return ((_exec, self.args), command_type)
+            return (self._raw, command_type)
```

Comment:
maybe no need to also store the `command_type` to make it even more simpler, as you already deduce from str in `__init__`